job.py

```diff
@@ -55,31 +55,26 @@ class Job:
     # These are dynamic, cached, and need to be udpated periodically
     phase = (None, None)   # Phase/subphase
 
-    def get_running_jobs(logroot):
-        '''Return the list of running plot jobs, reading the process table from scratch.'''
+    def get_running_jobs(logroot, cached_jobs=None):
```

Comment:
I think it could be a small bit tidier to just default this to an empty (immutable) collection.  Ought to drop the first `if` and the more complex if in the loop simplifies slightly.

```suggestion
    def get_running_jobs(logroot, cached_jobs=()):
```